scylladb/scylladb.go

```diff
@@ -0,0 +1,170 @@
+package scylladb
+
+import (
+	"errors"
+	"fmt"
+	"github.com/gocql/gocql"
+	"time"
+)
+
+// Storage interface that is implemented by storage providers
+type Storage struct {
+	session   *gocql.Session
+	tableName string
+
+	selectQuery string
+	insertQuery string
+	deleteQuery string
+	resetQuery  string
+}
+
+var (
+	checkSchemaMsg = "the `value` row has an incorrect data type. " +
+		"The message should be BLOB, but it is instead %s. This could lead to encoding-related issues if the database is not migrated (refer to https://github.com/gofiber/storage/blob/main/MIGRATE.md)"
+	createKeyspaceQuery = "CREATE KEYSPACE IF NOT EXISTS %s WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};"
+	dropQuery           = `DROP TABLE IF EXISTS %s.%s;`
+	createTableQuery    = `CREATE TABLE IF NOT EXISTS %s.%s (key TEXT PRIMARY KEY, value BLOB)`
+	checkSchemaQuery    = `SELECT type FROM system_schema.columns WHERE keyspace_name = '%s' AND table_name = '%s' AND column_name = 'value';`
+	keyspaceMsg         = `Keyspace cannot be empty.`
+)
+
+// New creates a new storage
+func New(config ...Config) *Storage {
+	var err error
+	var session *gocql.Session
+
+	// Set default config
+	cfg := configDefault(config...)
+
+	if cfg.Keyspace == "" {
+		panic(keyspaceMsg)
```

Comment:
I don't think this ever happens, the configDefault() call above will set the Keyspace to the default value if empty. One thing that could happen is if the user set the Keyspace to something like "   ".

Reply:
Added a comment above to fix that issue in configDefault()